File: note_view/lib/dto/note.dart
class Note {
  final String id;
  final String title;
  final String description;
  final String payload;
  final Map<String, String> attributes;

  Note(
      {required this.id,
      required this.title,
      required this.description,
      required this.payload,
      required this.attributes});

  factory Note.fromJson(Map<String, dynamic> json) {
    return Note(
        id: json['id'],
        title: json['title'],
        description: json['description'],
        payload: json['payload'],
        attributes: json['attributes']);
  }
}

File: DatabaseService/src/main/java/a/gleb/databaseservice/dao/Note.java
package a.gleb.databaseservice.dao;

import a.gleb.databaseservice.dao.dto.NoteDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.ZonedDateTime;

@AllArgsConstructor
@Data
@ToString
@Document(collection = "db_coll")
public class Note {

    @Id
    public String id;

    public String title;

    public String description;

    public String payload;

    public ZonedDateTime time;


    public static Note fromNoteDto(NoteDto noteDto){
        return new Note(
                noteDto.getId(),
                noteDto.getTitle(),
                noteDto.getDescription(),
                noteDto.getPayload(),
                ZonedDateTime.now()
        );
    }
}
